src/inmanta/protocol/methods_v2.py

```diff
@@ -244,3 +244,14 @@ def get_api_docs(format: Optional[str] = None) -> ReturnValue[Union[OpenAPI, str
        Get the OpenAPI definition of the API
        :param format: Use 'openapi' to get the schema in json format
     """
+
+
+@typedmethod(path="/agent/{name}/pause", operation="PUT", arg_options=methods.ENV_OPTS, client_types=["api"], api_version=2)
```

Comment:
I am not a big fan of using PUT for this, because it does not modify the complete state of the agent. I think in rest apis for actions it is better to either do:

POST /agent/{name}/{action} with action pause|unpause

OR
POST /agent/{name}?action=

But the latter we do not support in protocol I think